### DISCUSSION TITLE
TDUEX in OPL871

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#120] [General]
Wants to pull a whole slew of unknown decks.

modify,p=opl871,lo=e,z,c=TDUEX./*edit,TDUEX

Anyone else getting this?

**Comment by wschaub** [comment]
How are you trying to build it? the correct build procedure is in the INSTALL file.  

What you are probably seeing is TDUEX calling CYBIL common decks which are taken care of in the TDU build procedure.

it pulls down a multifile off the NOS PF tape which has the cybil compiler, cybil runtime relocatables, and cybil common deck OPL separated by an EOF, copies them out to individual local files and then proceeds to build TDU using the cybil common deck OPL with an *oplfile directive to modify. 

If you want to manually check the sources out of OPL871 you might do something like this:
reclaim.
?copy,fn=tdu1871,un=0
?
/rewind(*)
/skipf(tdu1871,2)
/copybf(tdu1871,tdtool3)
/attach,opl871
/modify,p=opl871
?*oplfile tdtool3
*EDIT TDUEX,TDUIN,TDUOUT,TDUTAB
?
/ascii
/scopy,compile

**Comment by KC5VDJ** [comment]
Thanks.  Was pulling the sources to get a better handle on things.

I get a kick out of the warning in the preface of the compass 3.6 manual, like 90+% of system utils seem to be in compass.

I am at a loss on one thing here, how do I prevent pulling in the common decks with each listing, so that the commons are extracted individually and not duplicated in each?

Also, I have had to get a little creative to get them into winblowz, primarily because of an invalid char error that pops on many when trying ftp (winblowz 10 standard client, no mode change).

**Reply by wschaub** [reply]
In order to just extract the source decks (no need to even do the *oplfile for a source run) 
I do something like this:
modify,p=filenamehere,c=0,s,f.
then I do scopy on the source file and I have a couple of perl scripts that I use to split the decks out into individual files. 

If you want the ascii stuff like the TDU decks to be readable when you extract them you will want to use the ascii command first.
This all assumes you're using IAF. 

the c=0 means don't make a compile file, and S means create a SOURCE file without any of the various source directives interpreted.

You can also produce a SOURCE file that has the sequence numbers in it by giving the *sorseq directive to the modify prompt

I haven't found a way to do that for UPDATE, but the process is very similar and I have a script for that too in the zip file. 

My preferred way to copy the source file out is via a terminal buffer that's been set to infinite size and then just saving that to a file to be worked on by my perl scripts.

Here's a copy of my current perl scripts. you just pipe the file into standard input of the script. 
[splittools.zip](https://github.com/kej715/DtCyber/files/14072999/splittools.zip)


Most of the stuff in OPL871 is COMPASS, but there are a few things in SYMPL and CYBIL. 

A lot of the product set is in SYMPL and most of the product set are in UPDATE format program libraries.  I've actually extracted most of NOS and the product set already so if you want to save some of the trouble you can look here:

https://drive.google.com/drive/folders/1s5Nts1yey0S8LHJZoIu5IgHLmVpefk4g?usp=drive_link

**Comment by KC5VDJ** [comment]
That will save a bit of time, thank you!

More on that ftp thing, and it is throwing me a bit.

This was using `type ascii`, and it happens with a large percentage of files.

On a side note here, the cols 73-EOL sequencing sometimes begins well to the left of col 73 on occasion.  Should those be corrected?

![LvCNrGUonJ_110342](https://github.com/kej715/DtCyber/assets/136194499/f6abece6-f910-4979-aad2-7c95f948cf9e)
![5k6jxpFwaG_110332](https://github.com/kej715/DtCyber/assets/136194499/89d53c9e-8b5c-478e-abcc-219f1f653092)

**Reply by wschaub** [reply]
I'm not sure about the FTP problem. I don't use the FTP server much I might experiment with it a bit.

**Reply by KC5VDJ** [reply]
At first I thought it was choking on eor/eof marks or something, but where it seems to happen seems quite random, but always at the same spots.

**Comment by kej715** [comment]
By default, the FTP server assumes that the files you’re trying to download are encoded in 6/12 Extended Display Code (I.e., mixed case). If the file is not actually 6/12. Then the FTP server might encounter invalid character encodings. A COMPASS program listing definitely won’t be 6/12. It will be 6-bit Display Code, so it should be downloaded as in:  ftp> get cybfile/cp=dis pcfile.txtSee the TCP/IP User Guide for details about FTP usage.On Jan 27, 2024, at 2:56 PM, William Schaub ***@***.***> wrote:﻿
I'm not sure about the FTP problem. I don't use the FTP server much I might experiment with it a bit.

**Reply by KC5VDJ** [reply]
200 COMMAND OKAY.
501 SYNTAX ERROR IN PARAMETERS OR ARGUMENTS.
ftp>

Maybe this is partially a winblowz thing? Most clients I have tried seem to have difficulty with nos, but iirc, many always did; there used to be a few (not many) public ftp servers on cybers back in the day, but I know they worked with, iirc, sunos, thus what? 4.3 Net 2?  Iirc, before Solaris, sunos was largely 4.x based.  As such, i recall it working, wonkily, with cybers.  Figures, who was in the room in the IEN meeting minutes?

I may boot into Linux and try.

I will end up writing a Directory Opus plugin for this.  Looking at dealing with ftp anyway for an elisp emacs mode (think ange-ftp) for using a real editor.  Granted, there is a beauty to the efficiency of xedit and other line editors when compared to the 'anything goes' full screen editors.  Had to think more about what you were doing, mistakes being more costly in time/effort.

**Comment by kej715** [comment]
Sorry, there was a typo in my reply. The “get” command should be:    get cybfile/cs=dis pcfile.txt“cs” stands for “character set”.  Other attributes may be provided as well. For example, on a “put” command, you may specify “ia” or “da” to indicate whether you want the uploaded file to be stored as indirect access or direct access. The default is direct access. So, I often specify “put” as in:   put pcfile.txt cybfile/ia,cs=disCheers,KevinOn Jan 28, 2024, at 12:15 PM, KC5VDJ ***@***.***> wrote:﻿
200 COMMAND OKAY.
501 SYNTAX ERROR IN PARAMETERS OR ARGUMENTS.
ftp>
Maybe this is partially a winblowz thing? Most clients I have tried seem to have difficulty with nos, but iirc, many always did; there used to be a few (not many) public ftp servers on cybers back in the day, but I know they worked with, iirc, sunos, thus what? 4.3 Net 2?  Iirc, before Solaris, sunos was largely 4.x based.  As such, i recall it working, wonkily, with cybers.  Figures, who was in the room in the IEN meeting minutes?
I may boot into Linux and try.

**Reply by KC5VDJ** [reply]
Yup, as I thought, it *IS* a winblowz thing.  No probs at all doing it straight, and with cs and output file specified, both working as they should, in MiNT.

M$ needs to update their ftp client...broken.  Directory Opus 12 (and prob the new 13 as well) will log you in, try to read the directory, then bails out completely back to an unconnected ftp pane.  The latter client is worth upgrading and adding a plugin (any sense in doing it for 12 at this point?  Still, there are probably many who have yet to upgrade, dopus works so well).

**Reply by KC5VDJ** [reply]
P.S. I don't work for the dopus people.  I just use their product, and find it impossible to replace.  They are kind of in a space by themselves.  Think Norton Commander on steroids for modern stuff, with an Amiga lineage, vastly configurable and expandable.
